ambuild2/frontend/paths.py

```diff
@@ -50,6 +50,13 @@ def Join(*nodes):
   return os.path.join(*paths)
 
 def IsSubPath(other, folder):
-  folder = os.path.normpath(folder) + os.sep
-  other = os.path.normpath(other) + os.sep
-  return other.startswith(folder)
+  other = os.path.abspath(other)
+  folder = os.path.abspath(folder)
+
+  relative = os.path.relpath(other, folder)
+  if relative.startswith(os.pardir): # Short-circuit on '..' -OR- '../', this is always pointing outside of "folder".
+    return False
+  elif relative.startswith(os.curdir): # Short-circuit on '.' -OR- './', this is always pointing to a child item.
+    return True
+
+  return other.startswith(folder) # In most cases, we shouldn't ever arrive to this point.
```

Comment:
At least `folder` should be suffixed with `os.sep`, so that `/path/foo` does not match `/path/foobar`, right?

Reply:
Local testing suggests that the short-circuits will handle it.
```
>>> print(foo_p1, foobar_p1, IsSubPath(foobar_p1, foo_p1))
L:\lobster\voided\foobar\base\lobster L:\lobster\voided\foo\base\lobster ..\..\..\foobar\base\lobster
lobster\voided\foo\base\lobster L:\lobster\voided\foobar\base\lobster False
```

Reply:
This whole module is begging for some unit testing. Anyway, this new logic looks broken to me:

```
>>> import os
>>> path1 = "C:\\x"
>>> path2 = "D:\\x"
>>> os.path.relpath(path1, path2)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "C:\Users\dvander\AppData\Local\Programs\Python\Python38-32\lib\ntpath.py", line 703, in relpath
    raise ValueError("path is on mount %r, start on mount %r" % (
ValueError: path is on mount 'C:', start on mount 'D:'
```